commcare_connect/opportunity/api/views.py

```diff
@@ -11,3 +14,20 @@ class OpportunityViewSet(viewsets.ReadOnlyModelViewSet):
 
     def get_queryset(self):
         return Opportunity.objects.filter(opportunityaccess__user=self.request.user)
+
+
+class UserLearnProgressView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def get(self, *args, **kwargs):
+        qs = OpportunityAccess.objects.all()
+        opportunity_access = get_object_or_404(qs, user=self.request.user, opportunity=kwargs.get("pk"))
```

Comment:
nit: can be simplified by just passing the class:
```suggestion
        opportunity_access = get_object_or_404(OpportunityAccess, user=self.request.user, opportunity=kwargs.get("pk"))
```